cpp/src/io/utilities/datasource.cpp

```diff
@@ -335,6 +335,13 @@ std::unique_ptr<datasource> datasource::create(host_buffer const& buffer)
     reinterpret_cast<const uint8_t*>(buffer.data), buffer.size));
 }
 
+std::unique_ptr<datasource> datasource::create(cudf::host_span<std::byte const> buffer)
```

Comment:
probably not relevant, but `datasource::create(host_buffer const& buffer)` could call this one instead of creating the arrow source directly.